common/dynamicconfig/constants.go

```diff
@@ -2133,6 +2133,16 @@ that task will be sent to DLQ.`,
 		10,
 		`Maximum number of resend events batch for a single replication request`,
 	)
+	WorkflowIdReuseMinimalInterval = NewNamespaceDurationSetting(
+		"history.workflowIdReuseMinimalInterval",
+		1*time.Second,
+		`WorkflowIdReuseMinimalInterval is used for timing how soon users can create new workflow with the same workflow ID.`,
+	)
+	EnableWorkflowIdReuseStartTimeValidation = NewNamespaceBoolSetting(
+		"history.enableWorkflowIdReuseStartTimeValidation",
+		false,
+		`If true, validate the start time of the old workflow is older than WorkflowIdReuseMinimalInterval when reusing workflow ID.`,
+	)
```

Comment:
Would it be simpler to set `history.workflowIdReuseMinimalInterval` to zero by default and treat that as "disabled"? That saves us from having another config variable.

Reply:
workflowIdReuseMinimalInterval is also used for protecting continue as new. We don't want to disable that.